services/skus/handler/handler.go

```diff
@@ -67,3 +77,55 @@ func (h *Order) Create(w http.ResponseWriter, r *http.Request) *handlers.AppErro
 
 	return handlers.RenderContent(ctx, order, w, http.StatusCreated)
 }
+
+func (h *Order) CreateNew(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+	raw, err := io.ReadAll(io.LimitReader(r.Body, reqBodyLimit10MB))
+	if err != nil {
+		return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+	}
+
+	req := &model.CreateOrderRequestNew{}
+	if err := json.Unmarshal(raw, req); err != nil {
+		return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+	}
+
+	ctx := r.Context()
+
+	if err := h.valid.StructCtx(ctx, req); err != nil {
+		verrs, ok := collectValidationErrors(err)
+		if !ok {
+			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+		}
+
+		return handlers.ValidationError("request body", verrs)
+	}
+
+	lg := logging.Logger(ctx, "payments").With().Str("func", "CreateOrderNew").Logger()
+
+	result, err := h.svc.CreateOrder(ctx, req)
+	if err != nil {
+		lg.Error().Err(err).Msg("failed to create order")
+
+		if errors.Is(err, model.ErrInvalidSKU) {
+			return handlers.WrapError(err, "Error validating SKU", http.StatusBadRequest)
+		}
+
+		return handlers.WrapError(err, "Error creating the order in the database", http.StatusInternalServerError)
```

Comment:
Given this is a new endpoint should we sanitise the error or do we need to maintain this for backward compatibility? The same could apply to the err invalid sku [here](https://github.com/brave-intl/bat-go/pull/1916/files#diff-91dbfa70ead67ea9d10e8ca048aeed2d41b1d47024e66737d6343c7519df28f8R110)

Reply:
It's a good question. The only supposed consumer of this endpoint currently _does not_ interpret in any way error responses from the existing status-quo endpoint. So I might as well think about it.

Reply:
Here and below, I've updated error messages.

As for the proper error sanitisation, I think I've got some ideas for how I'd want to see it in this code base, but the current situation seems rather limiting.

Do you think we could accept the updated wording here (and in other places)? The same question applies to the comment below.